### PR TITLE
Feature/parth4apple/user schema routes

### LINKED ISSUE #26: User schema and routes

### PR DESCRIPTION
**Pull request by parth4apple**
### Administrative Info
[GitHub Issue #]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- I added user schemas and routes.

### Testing
How did you confirm your changes worked? 
- I used postman to test creating a new user and logging in.

### CODE REVIEW

**Review by jacobayu — APPROVED** [review]
lgtm very cool

File: models/user.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    email: {
+      type: String,
+      required: true,
+      unique: true,
+    },
+    password: {
+      type: String,
+      required: true,
+    },
```
Comment (petabite) [inline models/user.js:28]: Since we are using Firebase for authentication now, I don't think we need to save passwords.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Looks nice Parth! There's a few changes needed:

1. In your validators variable, add a check to see if the last active date is in this format: "MM/DD/YY". Also, make sure to check that the email field is in email address format.
2. For all routes, when there's an error (when the user couldn't be validated, when the user couldn't be created, etc.), send a res status 500 message instead of throwing an error.
3. I realized we need one more route for users to retrieve a specific user by ID. Can you add this route? I'll add the details for this route in the Project Details doc.

**Review by petabite — APPROVED** [review]
👌

**Review by dnanjunda — APPROVED** [review]
LGTM!

**Pull request merged**